daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -87,13 +87,13 @@ final class ImmArray[+A] private (
   }
 
   /** O(1), crashes on empty list */
-  def head: A = this(0)
+  lazy val head: A = this(0)
```

Comment:
same story for this one and everything else in this class

Reply:
Same story here. It is not necessary to use extra memory to cache an O(1) operation.
Note that Scala immutable collections have `head`, `tail`, `isEmpty` and `nonEmpty`  as `def`.